### ISSUE TITLE
ModuleNotFoundError: No module named 'uvicorn' on SockJSAdapter.py

### ISSUE THREAD

**Issue by pevolution-ahmed** [rstudio/shiny-server#547]
### File in question:
 **SockJSAdapter.py**
### The Error Message:
`Using Python 3.10.6 at /usr/bin/python3
Traceback (most recent call last):
  File "/opt/shiny-server/python/SockJSAdapter.py", line 14, in <module>
    import uvicorn
ModuleNotFoundError: No module named 'uvicorn'`
### System Specs:
**Ubuntu 22.04**
### How to reproduce it:
- Just run shiny-server with Shiny for python application

**Comment by gdsutton** [comment]
Also getting this same error, did you manage to resolve it? Installing uvicorn through pip install in my dockerfile doesn't help as far as I can see

**Comment by avhz** [comment]
Anyone found a way to resolve this ?

**Comment by philibe** [comment]
TLDR: The cause is that shiny-server doesn't load its python environnment in particular `shiny` and  `uvicorn`.

- Test your app WITHOUT server, mono app with one port :
   + create python environment for shiny (if not yet done)
          - `conda create -n py-shinytest2 python=3.12.3`   ( also you can look at `conda install conda-forge::shiny`)
          - `conda activate py-shinytest2`
          - `pip install shiny` (verify before that `which pip` gives `/home/youruser/anaconda3/envs/py-shinytest2/bin/pip`)
   + install the demo of shiny for python
          - `mkdir -p /opt/shiny-server/samples/sample-apps/shiny-for-python`
          - `cd /opt/shiny-server/samples/sample-apps/shiny-for-python`
          - `shiny create --template dashboard-tips` then answer `Enter destination directory: ./dashboard-tips` and `Would you like to use Shiny Express? No` for example
          - `cd dashboard-tips`
          - `pip install -r requirements.txt`
   + launch of mono Shiny for Python Server for this demo
          - `shiny run app.py --port=3939 --host=0.0.0.0`
   + use of this  app
          - http://192.168.1.XXX:3939/ et voilà ! 


If  your app works.

- shiny WITH server, multi apps with one port
   + install of Shiny-Server 
          - Install shiny-server https://posit.co/download/shiny-server/ in summary `sudo gdebi  shiny-server-YY.YY.YY-amd64.deb` (I don't kown If the installation of R is mandatory if you will use only the Python part of Shiny Server, not R Shiny)
   + fast try for our  test to not have your pre existing R Shiny Server broken
          - `cp /etc/shiny-server/shiny-server.conf ~/shiny-server.TEST.conf` where you have to add a the beginning  `python /home/youruser/anaconda3/envs/py-shinytest2/bin/python;` (not forget the `python` word at the beginning of the line) and modify the port `listen 3939;`
          - test for conda but without shell conda init `source /home/youruser/anaconda3/bin/activate /home/youruser/anaconda3/envs/py-shinytest2/` => verify after  that `which python` gives `/home/youruser/anaconda3/envs/py-shinytest2/bin/python`
   + install the demo of shiny for python (see above)
   + launch of multi Shiny for Python Server for this demo ( to adapt from  `/etc/systemd/system/shiny-server.service` but for Shiny for Python with`pidfile` `ExecStart` etc )
          - `sudo /usr/bin/env bash -c 'source /home/youruser/anaconda3/bin/activate /home/youruser/anaconda3/envs/py-shinytest2/ && exec /opt/shiny-server/bin/shiny-server /home/youruser/shiny-server.TEST.conf'`
   + use of this app
          - http://192.168.1.220:3939/sample-apps/shiny-for-python/dashboard-tips et voilà !


If it works : 

- update your actual shiny-server installation with `/etc/systemd/system/shiny-server.service` and `/etc/shiny-server/shiny-server.conf` updated,
- or create a new service for shiny for python `/etc/systemd/system/shiny-server-python.service`  with a new `/etc/shiny-server/shiny-server-python.conf`. If you have a pre existing R  Shiny shiny-server, IMHO I will take this choice.

**Comment by philibe** [comment]
my `~/shiny-server.TEST.conf` for the test. 
```
python /home/youruser/anaconda3/envs/py-shinytest2/bin/python;

run_as shiny;

preserve_logs true;

server {
  listen 3939;
  location / {

    site_dir /srv/shiny-server;
    log_dir /var/log/shiny-server;
    directory_index on;
    sanitize_errors off;
    disable_protocols xdr-streaming xhr-streaming iframe-eventsource iframe-htmlfile;

  }
}
```
